app/helpers/maintenance_tasks/tasks_helper.rb

```diff
@@ -101,16 +101,38 @@ def csv_file_download_path(run)
       )
     end
 
-    # Return the appropriate field tag for the parameter, based on its type.
-    # If the parameter has a `validates_inclusion_of` validator, return a dropdown list of options instead.
-    def parameter_field(form_builder, parameter_name)
-      inclusion_validator = form_builder.object.class.validators_on(parameter_name).find do |validator|
+    # Returns the appropriate value for validator's inclusion options for a Proc,
+    # lambda, symbol, or other types.
+    #
+    # If there is a value returned, the dropdown list of options will be used.
+    #
+    # Procs, lambdas, and symbols options are not supported by the dropdown form,
+    # and return nil.
+    #
+    # @param object [Task] The object for which the value needs to be resolved.
+    # @param parameter_name [String] The parameter name.
+    #
+    # @return [Object] value of the resolved inclusion option.
+    def resolve_inclusion_value(object, parameter_name)
+      inclusion_validator = object.class.validators_on(parameter_name).find do |validator|
         validator.kind == :inclusion
       end
+      return unless inclusion_validator
 
+      in_option = inclusion_validator.options[:in] || inclusion_validator.options[:within]
+
+      return if in_option.respond_to?(:call) || in_option.respond_to?(:to_sym)
+
+      in_option
+    end
+
+    # Return the appropriate field tag for the parameter, based on its type.
+    # If the parameter has a `validates_inclusion_of` validator, return a dropdown list of options instead.
+    def parameter_field(form_builder, parameter_name)
+      inclusion_values = resolve_inclusion_value(form_builder.object, parameter_name)
       return form_builder.select(
-        parameter_name, inclusion_validator.options[:in], prompt: "Select a value"
-      ) if inclusion_validator
+        parameter_name, inclusion_values, prompt: "Select a value"
+      ) if inclusion_values
```

Comment:
Style nitpick, the `return` all fits into one line now, it will be a bit neater.

Reply:
Oh nice -- applied in https://github.com/Shopify/maintenance_tasks/pull/962/commits/af4e6b0ec4affe1f529043ec9ed93e0bb5f1167c